extensions/2.0/Khronos/KHR_texture_procedurals/README.md

```diff
@@ -0,0 +1,1397 @@
+
+* **Node Input and Output Ports** The interface for a node’s incoming data is declared through **input ports**, which may be spatially-varying or uniform. The interface for a node’s outgoing data is declared through **output ports**.  
+
+* There is a specific set of supported **Data Types**. Every port must have a data type.
+
+* A **Pattern** is a node that generates or processes simple scalar, vector, and color data, and has access to local properties of any geometry that has been bound to a given material.
+
+* A **Node Graph** is a directed acyclic graph (DAG) of nodes, which may be used to define arbitrarily complex generation or processing networks.  Node Graphs describe a network of "pattern" nodes flowing into shader inputs, or define a complex or layered node in terms of simpler nodes. The former is called a **compound nodegraph** and the latter a **functional nodegraph**.
+
+* Node Graph:
+  * **Inputs** are nodes that define the interface for a graph’s incoming data.
+  * **Outputs** are nodes that define the interface for a graph’s outgoing data.
+
+* A **Node Definition** is a specification for a unique function with a given set of inputs and outputs. The logic for each function is represented either as a `functional graph` or as code. 
+
+## Extension Declaration
+
+Usage is indicated by adding the `KHR_texture_procedurals` extension identifier to the `extensionsUsed` array.
+
+```json
+{
+    "extensionsUsed": [
+        "KHR_texture_procedurals"
+    ]
+}
+``` 
+with the extension object defined in the `extensions array:
+
+```json
+{
+    "extensions": {
+        "KHR_texture_procedurals": {
+            "mimetype": "application/mtlx+json;version=<MaterialX_version>",
+            "procedurals": []
+        }
+    }
+}
+```
+It is assumed that a mimetype is always required. As part of the mimetype a version `<MaterialX_version>` is specified. This is the version of the MaterialX library specification used when writing to glTF. The version is specified as a string in the form of `<major version>.<minor version>`. For example, if the MaterialX library version is 1.39, the mimetype would be `application/mtlx+json;version=1.39`.
+
+The correspond version is specified in a MaterialX XML document as follows:
+
+```xml
+<materialx version="1.39">
+</materialx>
+```
+
+The version in the mimetype __is not__ the extension version. If in the future the schema needs to be modified (perhaps due to changes in MaterialX) then a new extension version would be required.
+
+The `procedurals` array specifies the procedural graphs that are used in the glTF asset. 
+
+### Versioning
+
+Elements from different versions of MaterialX is disallowed. External tooling must handle this (e.g. by performing an "upgrade" operation).
+
+<img src="./figures//version_upgrade.svg" width=100%>
+
```

Comment:
This section can be removed since use of KHR_texture_procedurals + KHR_texture_procedurals_MaterialX_1.39 will imply MaterialX library used.